frontend/src/scenes/audit-logs/DetailFilters.tsx

```diff
@@ -0,0 +1,218 @@
+                const cleanScope = scope.replace(/([A-Z])/g, ' $1').trim() || 'General'
+
+                if (!sections.has(cleanScope)) {
+                    sections.set(cleanScope, new Set())
+                }
+
+                scopeData.fields.forEach((field) => {
+                    sections.get(cleanScope)!.add(field.name)
+                    if (cleanScope === 'General') {
+                        generalFields.add(field.name)
+                    }
+                })
+            })
+        }
+
+        return Array.from(sections.entries())
+            .sort(([a], [b]) => (a === 'General' ? -1 : b === 'General' ? 1 : a.localeCompare(b)))
+            .map(([title, fieldSet]) => {
+                const fields = Array.from(fieldSet)
+                    .filter((field) => title === 'General' || !generalFields.has(field))
+                    .sort()
+                    .map((field) => ({ value: field, label: field }))
+
+                return { title, options: fields }
+            })
+            .filter((section) => section.options.length > 0)
+    }
+
+    const fieldSections = buildFieldOptions()
+    const allFields = fieldSections.flatMap((section) => section.options)
+
+    const handleAddFilter = (fieldName: string | null): void => {
+        if (!fieldName) {
+            return
+        }
+        setFilters({
+            detail_filters: {
+                ...detailFilters,
+                [fieldName]: { operation: 'exact' as const, value: '' },
+            },
+        })
+    }
+
+    const handleUpdateFilter = (fieldPath: string, filter: DetailFilter | null): void => {
+        const newFilters = { ...detailFilters }
+        if (filter === null) {
+            delete newFilters[fieldPath]
+        } else {
+            newFilters[fieldPath] = filter
+        }
+        setFilters({ detail_filters: newFilters })
+    }
+
+    const handleRemoveFilter = (fieldPath: string): void => {
+        const newFilters = { ...detailFilters }
+        delete newFilters[fieldPath]
+        setFilters({ detail_filters: newFilters })
+    }
+
+    if (!availableFilters?.detail_fields || Object.keys(availableFilters.detail_fields).length === 0) {
+        return <></>
+    }
+
+    return (
+        <div className="flex flex-col gap-2">
+            <label className="block text-sm font-medium">Detail Filters</label>
+
+            <div className="flex flex-col gap-2">
+                {Object.entries(detailFilters).map(([fieldPath, filter]) => (
+                    <DetailFilterRow
+                        key={fieldPath}
+                        fieldPath={fieldPath}
+                        filter={filter}
+                        availableFields={allFields}
+                        onUpdate={handleUpdateFilter}
+                        onRemove={() => handleRemoveFilter(fieldPath)}
+                    />
+                ))}
+            </div>
+
+            <LemonSearchableSelect
```

Comment:
Also, the scroll feels broken, when you scroll to the bottom of the list, it jumps back to the top. It's more the comonent thatn your implementation i think but something ot look intop.